### ISSUE TITLE
const enums results in false positive with the rule "no-shadow"

### ISSUE THREAD

**Issue by onderceylan** [angular-eslint/angular-eslint#212]
Hi there! Thanks for this great migration schematic, it saves a lot of time!

I've had an issue with my migration due to default eslint rule "no-shadow". It should be replaced with "@typescript-eslint/no-shadow".

```{
  // note you must disable the base rule as it can report incorrect errors
  "no-shadow": "off",
  "@typescript-eslint/no-shadow": ["error"]
}
```

Reference: https://github.com/typescript-eslint/typescript-eslint/issues/2484

**Comment by JamesHenry** [comment]
Thanks @onderceylan please could you also report this to https://github.com/typescript-eslint/tslint-to-eslint-config

That is what determines the config conversion logic.

I will leave this open here though because we will also need to update our dependency on that library when it is resolved, and update the configs and snapshots.

**Comment by onderceylan** [comment]
Hey @JamesHenry, thanks for your reply. I created an issue in the **eslint-to-eslint-config** repo as you suggested: https://github.com/typescript-eslint/tslint-to-eslint-config/issues/856

**Comment by JamesHenry** [comment]
This is eligible to be fixed in the v2 release because it requires updating `tslint-to-eslint-config` to the version that we will be using to apply the breaking change in #245

**Comment by inorganik** [comment]
I've also noticed the camelCase rule prevents you from naming enums with capital letters [as they generally are](https://www.typescriptlang.org/docs/handbook/enums.html) in typescript, e.g:

```ts
enum Direction {
  Up,
  Down,
  Left,
  Right,
}
```
maybe that's a separate issue though. Really appreciate this project!
